core/src/com/mikroysoft/game/FireEngine.java

```diff
@@ -31,26 +34,72 @@ public FireEngine(int MAPWIDTH, int MAPHEIGHT) {
         position = new Coordinate(300,300);
         projectiles = new ArrayList < Projectile> ();
         direction = 0;
-        speed = 2;
+        speed = 0;
+        maxSpeed = 0;
+        acceleration = 0;
         health = 100;
         fuel = 100;
         distanceTravelled = 0;
         maxFuel = 100;
         maxHealth = 100;
+        maxVolume = 100;
+        waterVolume = 100;
     }
 	
+    public boolean isMaxSpeed() {
+    	if(this.speed == this.maxSpeed) {
+    		return true;
+    	} else {
+    		return false;
+    	}
+    }
+    
+    public void increaseSpeed() {
+    	this.speed = this.speed + this.acceleration;
+    }
+    
+    public void resetSpeed() {
+    	this.speed = 0;
+    }
+    public void setSpeedAndAcceleration(float s, float a) {
```

Comment:
Why do we have a unique function to set speed and acceleration? How do we justify it in the architecture?